packages/components/src/components/file/file.component.ts

```diff
@@ -300,6 +301,56 @@ export default class SynFile extends SynergyElement implements SynergyFormContro
     this.files = files;
   }
 
+  private async handleTransferItems(items: DataTransferItemList | null): Promise<FileList> {
+    if (!items || items.length !== 1) {
+      this.value = '';
+      return new Promise((_resolve, reject) => { reject(new Error('No proper items found')); });
+    }
+
+    const itemsArray = Array.from(items);
+
+    const entries: (FileSystemEntry | null)[] = itemsArray.map((item) => item.webkitGetAsEntry());
+
+    if (!entries[0]?.isDirectory) {
+      this.value = '';
+      return new Promise((_resolve, reject) => { reject(new Error('Not a folder')); });
+    }
+
+    const filesPromises = entries.map(entry => this.getFilesFromEntry(entry));
+    const filesArray = await Promise.all(filesPromises);
+    const files = filesArray.flat();
+
+    function arrayToFileList(f: File[]): FileList {
```

Comment:
Inlining this would prevent the creation of a needless closure and get rid of unneeded indentation:


```suggestion
    const dataTransfer = new DataTransfer();
    Array.from(files).forEach(f => dataTransfer.items.add(f));
    return dataTransfer.files;
```

You could also return the dataTransfer object itself, which would make it possible to keep the descructuring in handleDrop like this:

```javascript
  private async handleDrop(e: DragEvent) {
    e.preventDefault();
    e.stopPropagation();

    if (!e.dataTransfer) {
      return;
    }

    const { files } = this.webkitdirectory
      ? await this.handleTransferItems(items)
      : e.dataTransfer;

```